src/lib/operators/aggregate.cpp

```diff
@@ -271,21 +284,29 @@ std::shared_ptr<const Table> Aggregate::_on_execute() {
   */
 
   // Allocate a temporary memory buffer, for more details see aggregate.hpp
+  // This calculation assumes that we use VariableLengthAggregateKeys - other data structures use less space, but
+  // that is fine
   size_t needed_size_per_aggregate_key =
       aligned_size<AggregateKey>() + _groupby_column_ids.size() * aligned_size<AggregateKeyEntry>();
-  size_t needed_size = aligned_size<KeysPerChunk>() + input_table->chunk_count() * aligned_size<AggregateKeys>() +
+  size_t needed_size = aligned_size<KeysPerChunk<AggregateKey>>() +
+                       input_table->chunk_count() * aligned_size<AggregateKeys<AggregateKey>>() +
                        input_table->row_count() * needed_size_per_aggregate_key;
+  needed_size *= 1.1;  // Give it a little bit more, just in case
 
   auto temp_buffer = boost::container::pmr::monotonic_buffer_resource(needed_size);
-  auto allocator = AggregateKeysAllocator{PolymorphicAllocator<AggregateKeys>{&temp_buffer}};
+  auto allocator = AggregateKeysAllocator{PolymorphicAllocator<AggregateKeys<AggregateKey>>{&temp_buffer}};
+  allocator.allocate(1);  // Make sure that the buffer is initialized
   const auto start_next_buffer_size = temp_buffer.next_buffer_size();
 
   // Create the actual data structure
-  auto keys_per_chunk = KeysPerChunk{allocator};
+  auto keys_per_chunk = KeysPerChunk<AggregateKey>{allocator};
   keys_per_chunk.reserve(input_table->chunk_count());
-
   for (ChunkID chunk_id{0}; chunk_id < input_table->chunk_count(); ++chunk_id) {
-    keys_per_chunk.emplace_back(input_table->get_chunk(chunk_id)->size(), AggregateKey(_groupby_column_ids.size()));
+    if constexpr (std::is_same_v<AggregateKey, pmr_vector<AggregateKeyEntry>>) {
```

Comment:
Replace type `pmr_vector<AggregateKeyEntry>` with defined type `VariableLengthAggregateKey`.

Reply:
Good point - I forgot that it was defined. To be honest though, I would rather remove `VariableLengthAggregateKey`. I think there is a greater value in directly seeing if we are talking about a single value, a fixed-size array, or a vector. What do you think?

Reply:
I think the data structure is self-explaining so that the additional type is not necessary.